src/uu/tail/src/args.rs

```diff
@@ -300,6 +301,11 @@ impl Settings {
         self.inputs.iter().all(|input| input.is_stdin())
     }
 
+    pub fn stdin_is_regular_file(&self) -> bool {
+        let metadata = fs::metadata("/dev/stdin");
```

Comment:
Maybe this is ok, but I think we see some occasional issues doing this sort of things on Mac OS X: #7583.

Can you maybe add comment in this function to say something like
`// TODO: Consider calling fstat on stdin fd instead (#7583)` ?